code-actions-sample/src/diagnostics.ts

```diff
@@ -23,21 +23,27 @@ export function refreshDiagnostics(doc: vscode.TextDocument, emojiDiagnostics: v
 
 	for (let lineIndex = 0; lineIndex < doc.lineCount; lineIndex++) {
 		const lineOfText = doc.lineAt(lineIndex);
-		if (lineOfText.text.includes(EMOJI)) {
-			diagnostics.push(createDiagnostic(doc, lineOfText, lineIndex));
+		const regex = RegExp(EMOJI, 'g');
+		const matches = lineOfText.text.matchAll(regex);
+		for (const arr of matches) {
+			arr.index !== undefined && diagnostics.push(
```

Comment:
This is trying to be too clever. Just use `if`

Reply:
When would `index` ever be `undefined` anyway? I couldn't find any documentation about that being the case for a match.